src/harness/reference_models/interference/aggregate_interference.py

```diff
@@ -0,0 +1,366 @@
+
+  if len(grants_inside) > 0:
+    for channel in channels:
+      # Get protection constraint over 5MHz channel range
+      protection_constraint = interf.ProtectionConstraint(latitude=protection_point[0],
+                                longitude=protection_point[1], low_frequency=channel[0],
+                                high_frequency=channel[1], entity_type=protection_ent_type)
+
+      # Identify CBSD grants overlapping with frequency range of the protection entity 
+      # in the neighborhood of the protection point and channel 
+      neighborhood_grants = interf.findOverlappingGrants(
+                              grants_inside, protection_constraint)
+
+      if not neighborhood_grants:
+        continue
+     
+      cbsd_interference_list = []
+
+      for grant in neighborhood_grants:
+        cbsd_interference_list.append(interf.computeInterference(
+          grant, grant.max_eirp, protection_constraint, fss_info, 
+          esc_antenna_info, region_type))
+
+      aggr_interference = convertAndSumInterference(cbsd_interference_list)
+
+      # Update aggregate interference for a protection constraint
+      aggregate_interference.SetAggregateInterferenceInfo(protection_point[0],
+            protection_point[1], aggr_interference)
+   
+  return  
+
+def calculateAggregateInterferenceForFssCochannel(fss_record, cbsd_list):
+  """Calculates aggregate interference for FSS co-channel.
+  
+  Calculates aggregate interference for each protection constraint in the FSS
+  co-channel based on the interferences caused by CBSD's grants in the 
+  neighborhood of the FSS co-channel protection entity.
+
+  Args:
+    fss_record: FSS co-channel protection entity
+    cbsd_list: list of CBSD objects containing registrations and grants
+  Returns:
+    Aggregate interference to FSS co-channel in the nested dictionary format.
+    {latitude : {longitude: [aggr_interf1(mW), aggr_interf2(mW)]}}
+  """
+  grant_list = interf.getAllGrantInformationFromCbsdDataDump(cbsd_list)
+
+  # Get the protection point of the FSS
+  fss_point = fss_record['record']['deploymentParam'][0]['installationParam']
+  protection_point = (fss_point['latitude'], fss_point['longitude'])
+
+  # Get the frequency range of the FSS
+  fss_freq_range = fss_record['record']['deploymentParam'][0]\
+                     ['operationParam']['operationFrequencyRange']
+  fss_low_freq = fss_freq_range['lowFrequency']
+
+  # Get FSS information
+  fss_info = interf.FssProtectionPoint(latitude=fss_point['latitude'], 
+               longitude=fss_point['longitude'],
+               height_agl=fss_point['height'],
+               max_gain_dbi=fss_point['antennaGain'],
+               pointing_azimuth=fss_point['antennaAzimuth'],
+               pointing_elevation=fss_point['antennaDowntilt'])
+
+  # Get channels for co-channel CBSDs
+  protection_channels = interf.getProtectedChannels(fss_low_freq, interf.CBRS_HIGH_FREQ_HZ)
+
+  aggregate_interference.__init__()
```

Comment:
Why this line ?  This is not the proper way `__init__` is the constructor of the object, and the object is already created.
The intent is to return a fresh aggreg object to the caller. Why not creating one here and returning it  in that case?

Also same issue on the other routines

Reply:
Removed reinitialization of global object, instead of that object is   created inside the function.           Added custom manager to share object across processes.